sematic/plugins/abstract_external_resource.py

```diff
@@ -207,6 +208,16 @@ def __post_init__(self):
         if not isinstance(self.status, ResourceStatus):
             raise ValueError(f"ExternalResource had invalid status: '{self.status}'")
 
+    @classmethod
+    def get_settings_vars(cls) -> Type[AbstractPluginSettingsVar]:
+        """
+        Returns the Settings var enum for this plug-in.
+
+        The class must inherit from `AbstractPluginSettingsVar` and list all
+        available settings for this plug-in.
+        """
+        return AbstractPluginSettingsVar
```

Comment:
Isn't this already defined on `AbstractPlugin`? Also, why doesn't `AbstractExternalResource` inherit from `AbstractPlugin`?

Reply:
It was originally created before `AbstractPlugin` existed, and also I was unsure about a dataclass inheriting from a non-dataclass. Seems to work though; changed it.